### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by diarisdiakite**
In this PR, I:
✅ Added the Redux Toolkit (`npm install react-redux @reduxjs/toolkit`).
✅ Created a directory that contains all the Redux logic (`/src/redux`)
✅ Configured a Redux store (`/src/redux/store.js`)
✅ Defined a `slice` of state for `books` that:
   - Includes an **array of books** (initial state: empty array)
   - Includes a **reducer that adds a book**
   - Includes a **reducer that removes a book**
✅ Define a `slice` of state for `categories` that:
   - Includes an **array of categories** (initial state: empty array)
   - Includes a **reducer that checks the status** and always returns "Under construction" (the initial state should check to that string)
✅ Structure my application files using a "feature folder" approach and used the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for the Redux files in the redux folder looking like this:
   ```sh
   /src
       |---/components
       |---/redux
           |--- /books
                   | booksSlice.js
           |--- /categories
                   | categoriesSlice.js
           | store.js
       | index.js
   ```

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      }
+    },
+    checkCategory: (state, action) => {
+      action.payload = 'Under construction';
+      return action.payload;
+      /*
+      state.categories = 'Under construction';
+      return state.categories */
+    },
+  },
```
Comment (AbelG101) [inline src/redux/categories/categoriesSlice.js]: - [ ] As part of the project requirements you only need 1 reducer which you already have the `checkCategory` and make sure that the reducer checks the status from the action payload before returning the message as shown below 👇.
```jsx
reducers: {
    checkCategory: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },

```

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @diarisdiakite 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Proper use of Redux ✔
- Correct implementation of Reducers and slices ✔
- Correct use of Git Flow ✔
- Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with an amazing summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (diarisdiakite) [reply src/redux/categories/categoriesSlice.js]: Thank you @AbelG101. I made the changes. Should I really remove the other reducers?

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @diarisdiakite👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with redux. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Gitflow is followed. :heavy_check_mark:



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**